sherpa/astro/flux.py

```diff
@@ -160,8 +160,10 @@ def print_sample_result(title, arg):
         for x in oflxiflx:
             sf = numpy.sort(x)
             median = numpy.median(sf)
-            upconfidence = sf[(1.0 - myconfidence) * size - 1]
-            loconfidence = sf[myconfidence * size - 1]
+            upconfidence_index = int((1.0 - myconfidence) * size - 1)
```

Comment:
It would be cleaner from a tracking perspective to keep the change as a straight forward fix and then add your suggestion of using numpy.percentile as a separate request. Doing so would allow more concise testing and reduce the risk of inadvertent side effects (subtleties) being introduced.